db/migrate/20251216122223_change_id_sequences_to_bigint.rb

```diff
@@ -0,0 +1,22 @@
+class ChangeIdSequencesToBigint < ActiveRecord::Migration[7.0]
+  # The 20170110113824_change_id_value_range migration changed logs.id, reports.id,
+  # and fact_values.id columns to bigint, but ActiveRecord's change_column doesn't
+  # update the underlying sequence type. This fixes the sequences to allow values
+  # beyond 2^31-1.
+
+  def up
+    change_sequence :logs_id_seq, as: :bigint
+    change_sequence :reports_id_seq, as: :bigint
+    change_sequence :fact_values_id_seq, as: :bigint
+  end
+
+  def down
+    raise ActiveRecord::IrreversibleMigration
+  end
+
+  private
+
+  def change_sequence(name, as:)
+    connection.execute("ALTER SEQUENCE #{connection.quote_table_name(name)} AS #{as}")
```

Comment:
I don't think there's a guarantee that the table name is the same as the sequence. In the ActiveRecord code I see you can at least use `connection.default_sequence_name(table_name, primary_key)` with the latter argument being optional. https://api.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/DatabaseStatements.html#method-i-default_sequence_name doesn't have a good description, but for PostgreSQL it'll do the right thing: return the sequence name for a column.

So I'd suggest:
```suggestion
  def change_sequence(table_name, as:)
    connection.execute("ALTER SEQUENCE #{connection.default_sequence_name(table_name)} AS #{as}")
```

Reply:
@ekohl 
Table name can be different than sequence. then we can use below to fetch sequence
```
# pg_get_serial_sequence queries the database for the real sequence name
sequence_name = conn.select_value(
  "SELECT pg_get_serial_sequence(#{conn.quote('logs')}, 'id')"
)

# Remove schema prefix if present (e.g., "public.logs_id_seq" -> "logs_id_seq")
sequence_name = sequence_name.split('.').last if sequence_name&.include?('.')
```
doc: **pg_get_serial_sequence**
https://www.postgresql.org/docs/16/functions-info.html#FUNCTIONS-INFO-CATALOG-TABLE-PG-GET-SERIAL-SEQUENCE

default_sequence_name will add `_id_seq` to `table name`.
```
[3] pry(main)> conn.default_sequence_name(:logs)
=> "public.logs_id_seq"
[4] pry(main)> conn.default_sequence_name(:logs_id_seq)
=> "logs_id_seq_id_seq"
```

What do you think?